Yoast/wordpress-seo
# Add backward compatibility for protected methods & properties to  to the deprecated schema classes

## Context


* We still want to support the code that extends these schema classes directly (until we remove the deprecated code).

## Summary


This PR can be summarized in the following changelog entry:

* Adds backward compatibility for protected methods & properties to the deprecated schema classes.

## Relevant technical choices:

* Extending the new classes (after checking with Herre if this was an okay solution).
* Trying to still call methods that disappeared best I could:
  - `Author->is_post_author` contents changed to match the if statement in `is_needed`.
  - `FAQ`
    * `__construct` and `prepare_schema` use filters that we have implemented differently now, but still should be used and effective.
    * `render_schema_questions` could maybe be recoded to `generate_question_block` but that seems not worth it right now.
    * Questions and Question_List are not actually implemented either, just should not produce fatals.
  - `HowTo`
    * Refactored the new HowTo generation into a protected method (`add_how_to`) that is similar as the old `render`.

## Test instructions

This PR can be tested by following these steps:

Compare [old code](https://github.com/Yoast/wordpress-seo/blob/3c927425518c3529125b91a04e57e6bcde465a98/frontend/schema/) to the current and see that I did not skip any `public` or `protected` methods or properties.
Pay some extra attention to that they are actually still used too.

Not really testing the protected functionality here. As this would require writing your own implementation. So please see the appropriate test instructions in this PR https://github.com/Yoast/wordpress-seo/pull/14583

### HowTo
Extra step since I refactored the new HowTo too.
* Please compare the schema output of a HowTo page to the output before the indexables (13.4 release?). It should be the same.

## UI changes

## Documentation

## Quality assurance


Fixes #14667
